### PR TITLE
Add ColocSettingsDialog to pointwiseColoc

### PR DESCRIPTION
**Pull request by zacsimile**
Accounts for images with … 3+ color channels

Addresses issue #622.

**Is this a bugfix or an enhancement?**
Yes

**Checklist:**

- [ ] Tested with numpy=1.14
- [ ] Tested on python 2.7 and 3.6
- [ ] Tested with wx=3.x and wx=4.x [if UI code]
- [ ] Does the PR avoid variable renaming in existing code, whitespace changes, and other forms of tidying? [There is a place for code tidying, but it makes reviewing 
much simpler if this is kept separate from functional changes]

If an enhancement (or non-trivial bugfix):

- [ ] Has this been discussed in advance (feature request, PR proposal, email, or direct conversation)?
- [ ] Does this change how users interact with the software? How will these changes be communicated?
- [ ] Does this maintain backwards compatibility with old data?
- [ ] Does this change the required dependencies?
- [ ] Are there any other side effects of the change?

### CODE REVIEW

**Comment by zacsimile** [comment]
One problem with this is it will not work if a user tries this with a single color channel. Ideally we'd show a dialog in `OnPointwiseColoc`

```
if len(chans) == 0:
            import wx
            md = wx.MessageDialog(self.visFr, 'Not enough colour channels', 'Pointwise colocalization requires 2 colour channels', wx.OK)
            md.ShowModal()
            return
```

But this will still cause the recipe to execute, so there won't be a rollback. I could add a `raise` in there, but that will generate two popups. Is there a good way to alert a user to a problem with a recipe and then rollback the recipe (not add it to the pipeline) without throwing an error in the context manager?

File: PYME/LMVis/Extras/pointwiseColoc.py [hunk 1]
```diff
@@ -21,8 +21,18 @@ def OnPointwiseColoc(self, event):
         #from PYME import mProfile
         #mProfile.profileOn(['distColoc.py'])
         from PYME.Analysis.Colocalisation import distColoc
+        from PYME.DSView.modules.coloc import ColocSettingsDialog
+
+        chans = self.visFr.pipeline.colourFilter.getColourChans()[::1]
```
Comment (David-Baddeley) [inline PYME/LMVis/Extras/pointwiseColoc.py]: What is the `[::1]` doing?

File: PYME/LMVis/Extras/pointwiseColoc.py [hunk 2]
```diff
@@ -21,8 +21,18 @@ def OnPointwiseColoc(self, event):
...
+
+        dlg = ColocSettingsDialog(self.visFr, names=chans, show_bins=False)
+        dlg.ShowModal()
```
Comment (David-Baddeley) [inline PYME/LMVis/Extras/pointwiseColoc.py]: Check return and abort if user cancelled

**Comment by David-Baddeley** [comment]
not sure what the problem with the single colour check above is ... As it stands, this doesn't add anything to the pipeline.

File: PYME/LMVis/Extras/pointwiseColoc.py (hunk 1 above)
Comment (zacsimile) [reply PYME/LMVis/Extras/pointwiseColoc.py]: Not sure. It's a historic thing that I just copied over.

**Comment by zacsimile** [comment]
You're right. I must have run something else before this.

File: PYME/LMVis/Extras/pointwiseColoc.py (hunk 2 above)
Comment (zacsimile) [reply PYME/LMVis/Extras/pointwiseColoc.py]: Weirdly returns `ID_OK` if you click `OK` or close the dialog. Have you run into this before, @David-Baddeley? I can't see a single good reason for this.

**Pull request merged**